golang/synthesiserConnector/main.go

```diff
@@ -0,0 +1,397 @@
+package main
+
+import (
+	"encoding/json"
+	helpers "iotics-training/helpers"
+	"log"
+	"net/url"
+	"time"
+
+	ioticsApi "github.com/Iotic-Labs/iotics-api-grpc-go/iotics/api"
+	"github.com/Iotic-Labs/iotics-identity-go/pkg/register"
+	"google.golang.org/protobuf/types/known/timestamppb"
+)
+
+const (
+	RESOLVER_URL  = ""
+	HOST_GRPC_URL = "" // <spacename>.iotics.com:10001
+
+	USER_KEY_NAME = ""
+	USER_SEED     = ""
+
+	AGENT_KEY_NAME = ""
+	AGENT_SEED     = ""
+)
+
+/*** BONUS ***/
+// func getInputMessages(inputListener ioticsApi.InputAPI_ReceiveInputMessagesClient, inputValueLabel string, ch chan bool) {
+// 	log.Print("Waiting for Input Messages...")
+// 	mappedData := make(map[string]interface{})
+
+// 	for {
+// 		msg, err := inputListener.Recv()
+// 		if err != nil {
+// 			log.Printf("Error: %v", err)
+// 			break
+// 		}
+// 		msgReceived := msg.GetPayload().Message.Data
+// 		err = json.Unmarshal(msgReceived, &mappedData)
+// 		if err != nil {
+// 			log.Printf("Error: %v", err)
+// 			break
+// 		}
+
+// 		presence := bool(mappedData[inputValueLabel].(bool))
+// 		log.Printf("Received Input message: %t", presence)
+
+// 		if presence {
+// 			ch <- true
+// 		} else {
+// 			ch <- false
+// 		}
+// 	}
+// }
+
+func getFeedData(feedListener ioticsApi.InterestAPI_FetchInterestsClient, twinId string, feedValueLabel string, ch chan bool) {
```

Comment:
could add channel direction here to be explicit - https://gobyexample.com/channel-directions